server/player.lua

```diff
@@ -580,259 +710,101 @@ function CreatePlayer(playerData, Offline)
     self.PlayerData = playerData
     self.Offline = Offline
 
+    ---@deprecated use UpdatePlayerData instead
     function self.Functions.UpdatePlayerData()
-        if self.Offline then return end -- Unsupported for Offline Players
-        TriggerEvent('QBCore:Player:SetPlayerData', self.PlayerData)
-        TriggerClientEvent('QBCore:Player:SetPlayerData', self.PlayerData.source, self.PlayerData)
+        assert(not self.Offline, 'unsupported for offline players')
```

Comment:
This is pretty much a breaking change as it will stop the caller's code from being executed, when previously it would continue. Consider switching this with a `lib.print.warn`.